### PR TITLE
#224 Download database backup from Google Drive

### PR DESCRIPTION
**Pull request by carlosmiranda**
#224: Download the database backup from Google Drive using `gdrive` CLI application.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#282](https://github.com/timelesslounge/timelessis/issues/282) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#282](https://github.com/timelesslounge/timelessis/issues/282) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/39d657fe-2eb3-42a3-aa85-605eb11dd5d6); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: scripts/backup/pg_restore.sh [hunk 1]
```diff
@@ -49,8 +47,12 @@ if [ "$BACKUP_USER" != "" -a "$(id -un)" != "$BACKUP_USER" ]; then
 fi;
 
 if [ -z $RESTORE_DIR ]; then
-	echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Backup not provided, please use -d option." 1>&2
-	exit 1;
+    echo -e "\nRestore directory is empty, retrieving from Google Drive backup"
+    if ! gdrive download "$DATABASE".sql.gz --timeout "$TIMEOUT"; then
```
Comment (vladarefiev) [inline scripts/backup/pg_restore.sh]: @carlosmiranda I think will be greate if we set a variable for path where download to, WDYT?

File: scripts/backup/pg_restore.sh [hunk 2]
```diff
@@ -49,8 +47,12 @@ if [ "$BACKUP_USER" != "" -a "$(id -un)" != "$BACKUP_USER" ]; then
 fi;
 
 if [ -z $RESTORE_DIR ]; then
-	echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Backup not provided, please use -d option." 1>&2
-	exit 1;
+    echo -e "\nRestore directory is empty, retrieving from Google Drive backup"
```
Comment (vladarefiev) [inline scripts/backup/pg_restore.sh]: @carlosmiranda BTW I didn't find where we install `gdrive`.

**Comment by vladarefiev** [comment]
@carlosmiranda looks good for me. but i've got two questions

File: scripts/backup/pg_restore.sh (hunk 2 above)
Comment (carlosmiranda) [reply scripts/backup/pg_restore.sh]: @vladarefiev there is info in `README.md`, added when #115 was implemented.

File: scripts/backup/pg_restore.sh (hunk 1 above)
Comment (carlosmiranda) [reply scripts/backup/pg_restore.sh]: @vladarefiev done

**Comment by carlosmiranda** [comment]
@vladarefiev please see latest commit. I made a further change after realizing Google Drive API needs file ID for download, and added a puzzle for a more graceful/user friendly implementation.

File: scripts/backup/pg_restore.sh (hunk 2 above)
Comment (vladarefiev) [reply scripts/backup/pg_restore.sh]: @carlosmiranda Ah, I thought it will be periodic task. But if it's manual I think it's ok.

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 26 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#282](https://github.com/timelesslounge/timelessis/issues/282) is now out of scope

**Comment by ypshenychka** [comment]
@vladarefiev According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Pull request merged**